lib/rust/lexer/definition/src/lexer.rs

```diff
@@ -90,89 +104,221 @@ impl EnsoLexer {
+            self.on_eof(reader);
+        }
     }
 
-    /// The characters that break token lexing in Enso.
-    fn break_chars() -> String {
-        let mut break_chars = String::from("`@#,. ");
-        break_chars.push_str(&Self::operator_chars());
-        break_chars.push_str(&Self::whitespace_break_chars());
-        break_chars.push_str(&Self::group_chars());
-        break_chars
+
+    // === Comment Rules ===
+
+    /// The rules for lexing Enso comments.
+    fn add_comment_rules(lexer:&mut EnsoLexer) {
+        let comment_char      = lexeme::comment().into_pattern();
+        let lf                = lexeme::lf().into_pattern();
+        let crlf              = lexeme::crlf();
+        let eof               = lexeme::eof();
+        let spaces            = lexeme::spaces();
+        let doc_comment_start = lexeme::doc_comment().into_pattern() >> &spaces.opt();
+        let eof_newline       = &spaces.opt() >> &eof;
+        let empty_line_lf     = &spaces.opt() >> &lf;
+        let empty_line_crlf   = &spaces.opt() >> &crlf;
+        let any               = Pattern::any();
+
+
+        // === Initial State Rules ===
+
+        let initial_state_id = lexer.initial_state;
+        let initial_state    = lexer.group_mut(initial_state_id);
+        rule!(initial_state(doc_comment_start) => self.on_begin_doc_comment(reader));
+        rule!(initial_state(comment_char)      => self.on_begin_disable_comment(reader));
+
+
+        // === Disable Comment Rules ===
+
+        let disable_comment_id = lexer.disable_comment;
+        let disable_comment    = lexer.group_mut(disable_comment_id);
+        rule!(disable_comment(lf)   => self.on_end_disable_comment(token::LineEnding::LF,reader));
+        rule!(disable_comment(crlf) => self.on_end_disable_comment(token::LineEnding::CRLF,reader));
+        rule!(disable_comment(eof)  => self.on_end_disable_comment(token::LineEnding::None,reader));
+        rule!(disable_comment(any)  => self.on_build_disable_comment(reader));
+
+
+        // === Doc Comment Rules ===
+
+        let doc_comment_id = lexer.doc_comment;
+        let doc_comment    = lexer.group_mut(doc_comment_id);
+        rule!(doc_comment(lf)   => self.on_doc_comment_end_of_line(token::LineEnding::LF,reader));
+        rule!(doc_comment(crlf) => self.on_doc_comment_end_of_line(token::LineEnding::CRLF,reader));
+        rule!(doc_comment(eof)  => self.on_doc_comment_end_of_line(token::LineEnding::None,reader));
+        rule!(doc_comment(any)  => self.on_build_doc_comment_line(reader));
+
+
+        // === Newline Handling in Doc Comments ===
+
+        let doc_comment_newline_id = lexer.doc_comment_newline;
+        let doc_comment_newline    = lexer.group_mut(doc_comment_newline_id);
+        rule!(doc_comment_newline(spaces.opt()) => self.doc_comment_on_new_line(reader));
+        rule!(doc_comment_newline(empty_line_lf) => self.doc_comment_on_empty_line(
+            token::LineEnding::LF,
+            reader
+        ));
```

Comment:
Write it shorter pls (we are never calling functions providing args line by line):

```rust
        rule!(doc_comment_newline(empty_line_lf) => self.doc_comment_on_empty_line
            (token::LineEnding::LF,reader));
```

or, alternatively:

```rust
        rule!(doc_comment_newline(empty_line_lf) => 
            self.doc_comment_on_empty_line(token::LineEnding::LF,reader));
```